### ISSUE TITLE
Help building dukpy filter

### ISSUE THREAD

**Issue by kevingill1966** [amol-/dukpy#38]
Hi,

I need help with building a dukpy filter. I am trying to build tools to create vue.js assets via webassets.
I copied the logic in the less filter. When I run the tests I just get:

      JSRuntimeError: SyntaxError: unterminated statement (line 2). 

However the file pointed to is the package.json file. The full code is in my fork at..

     https://github.com/kevingill1966/dukpy

Full traceback is in

      https://github.com/kevingill1966/dukpy/blob/master/PROBLEM.rst

Summary traceback...

  File "/srv/insight_nrn/submodules/dukpy/dukpy/webassets/vuejsfilter.py", line 24, in input                                                                                                                   
    src = dukpy.vue_template_compiler(_in.read(), options=options)                                                                                                                                             
  File "/srv/insight_nrn/submodules/dukpy/dukpy/vuejs.py", line 18, in vue_template_compiler                                                                                                                   
    options=options                                                                                                                                                                                            
  File "/srv/insight_nrn/submodules/dukpy/dukpy/evaljs.py", line 57, in evaljs                                                                                                                                 
    res = _dukpy.eval_string(self, jscode, jsvars)                                                                                                                                                             
JSRuntimeError: SyntaxError: unterminated statement (line 2)                                                                                                                                                   
    vue-template-compiler/package.json:2                                                                                                                                                                       
    duk_js_compiler.c:6594                                                                                                                                                                                     
    anon  native strict preventsyield
    anon vue-template-compiler:5 preventsyield
    require  native strict preventsyield
    eval src/pyduktape.c:1 preventsyield

**Comment by amol-** [comment]
The problem seems to be caused by those two lines of code: https://github.com/kevingill1966/dukpy/blob/master/dukpy/jsmodules/vue-template-compiler/index.js#L5-L6

But note that the engine used by dukpy is stricter than the nodejs engine regarding regular expressions so you will have to fix regular expressions in the codebase before you can make the compiler actually able to rung. Mostly you will have to escape all `[` literal characters with `\[`

**Comment by amol-** [comment]
FYI, I usually go for UMD versions of the modules when vendoring them in DukPy, they usually work more easily.

**Comment by amol-** [comment]
FYI, I was able to make the compiter run with following changes:

```
diff --git a/dukpy/jsmodules/vue-template-compiler/build.js b/dukpy/jsmodules/vue-template-compiler/build.js
index 0fc6b98..fbc98eb 100644
--- a/dukpy/jsmodules/vue-template-compiler/build.js
+++ b/dukpy/jsmodules/vue-template-compiler/build.js
@@ -422,7 +422,7 @@ function parseHTML (html, options) {
         if (!isPlainTextElement(stackedTag) && stackedTag !== 'noscript') {
           text = text
             .replace(/<!\--([\s\S]*?)-->/g, '$1') // #7298
-            .replace(/<!\[CDATA\[([\s\S]*?)]]>/g, '$1');
+            .replace(/<!\[CDATA\[([\s\S]*?)\]\]>/g, '$1');
         }
         if (shouldIgnoreFirstNewline(stackedTag, text)) {
           text = text.slice(1);
@@ -3388,7 +3388,7 @@ function isDirectChildOfTemplateFor (node) {
 /*  */
 
 var fnExpRE = /^([\w$_]+|\([^)]*?\))\s*=>|^function\s*\(/;
-var simplePathRE = /^[A-Za-z_$][\w$]*(?:\.[A-Za-z_$][\w$]*|\['[^']*?']|\["[^"]*?"]|\[\d+]|\[[A-Za-z_$][\w$]*])*$/;
+var simplePathRE = /^[A-Za-z_$][\w$]*(?:\.[A-Za-z_$][\w$]*|\['[^']*?'\]|\["[^"]*?"\]|\[\d+\]|\[[A-Za-z_$][\w$]*\])*$/;
 
 // KeyboardEvent.keyCode aliases
 var keyCodes = {
diff --git a/dukpy/jsmodules/vue-template-compiler/index.js b/dukpy/jsmodules/vue-template-compiler/index.js
index 287d281..3a3cc20 100644
--- a/dukpy/jsmodules/vue-template-compiler/index.js
+++ b/dukpy/jsmodules/vue-template-compiler/index.js
@@ -2,8 +2,8 @@ try {
   var vueVersion = require('vue').version
 } catch (e) {}
 
-var packageName = require('./package.json').name
-var packageVersion = require('./package.json').version
+var packageName = "vue-template-compiler"
+var packageVersion = "2.5.17"
 if (vueVersion && vueVersion !== packageVersion) {
   throw new Error(
     '\n\nVue packages version mismatch:\n\n' +
diff --git a/tests/test_webassets_filter.py b/tests/test_webassets_filter.py
index b8c49e1..0c5970c 100644
--- a/tests/test_webassets_filter.py
+++ b/tests/test_webassets_filter.py
@@ -161,6 +161,5 @@ class TestVueFilter(TempEnvironmentHelper):
     def test_template(self):
         self.create_files({'in': self.TEMPLATE})
         self.mkbundle('in', filters='vuejs-template', output='out').build()
-        import pdb; pdb.set_trace()
         assert '_createClass(TEST, ' in self.get('out')
         assert 'require' in self.get('out')
```

**Comment by kevingill1966** [comment]
Thanks Alessandro,

With respect to the first problem: https://github.com/kevingill1966/dukpy/blob/master/dukpy/jsmodules/vue-template-compiler/index.js#L5-L6

I found a related discussion in the duktape issues: https://github.com/svaarala/duktape/issues/1933.
The problem arises because dukpy uses "eval" mode and the code assumes "compiled" mode. I copied the .json file to a .js file and made it valid javascript by wrapping with ().

I applied your regular expression fixes.

This step is working now.
